src/wallet/db.cpp

```diff
@@ -20,6 +21,12 @@
 #include <boost/thread.hpp>
 
 namespace {
+
+struct BerkeleyFileid {
+    u_int8_t value[DB_FILE_ID_LEN];
+};
+std::unordered_map<std::string, BerkeleyFileid> g_fileids;
```

Comment:
I think it'd be more correct to add a `std::unordered_map<std::string, BerkeleyFileid> m_fileids` member in `BerkeleyEnvironment` member instead of a `g_fileids` global because there can be duplicate filenames if the files are in different directories.

Also, making this a member would be consistent with existing `mapFileUseCount` `mapDb` members there which are also maps indexed by filename. In the future it could be nice to consolidate the 3 maps.

Reply:
Updated